### PR TITLE
Search render prop

### LINKED ISSUE #92: Create `Downshift` style component for the search input
[`Downshift`][DownshiftGithub] is used internally for the `<Input />` component, and one of the things I really like about the [`Downshift`][DownshiftGithub] implementation is how it doesn't actually render anything itself, it passes through control of the HTML rendering to the end user, but abstracts and simplifies the business case of creating autocomplete/dropdown components.

It would be nice to abstract the business logic of our `<Input />` component into something similar to how [`Downshift`][DownshiftGithub] operates to make it easier for end users to provide their own markup and implementation of the search input.

Reference: [Downshift Usage](https://github.com/paypal/downshift#usage)

[DownshiftGithub]: https://github.com/paypal/downshift

### PR DESCRIPTION
**Pull request by zlatanpham**
resolve #92

### CODE REVIEW

**Comment by zlatanpham** [comment]
@benhinchley, just quickly do the replacement. I'll do document for the Search component in the next commit.

**Comment by zlatanpham** [comment]
Added document for `Search`

File: src/components/Search/search.mdx [hunk 1]
```diff
@@ -0,0 +1,235 @@
...
+- `paginate`: `(page: number) => void`
+- `search`: `(query: string, override: boolean) => void`
+- `instantSearch`: `(query: string, override: boolean) => void`
+
+They are available to use when call:
+
+- Render function: `<Search>{searchProps => <div>/* your JSX here! */</div>}</Search>`
+- Props: `stateReduder` or `onChange`.
+
+## Usage
```
Comment (benhinchley) [inline src/components/Search/search.mdx]: I know these are examples, but if we could clean up the styling just a tad to make them look a little nicer that would be great

File: src/components/Input/Input.tsx [hunk 2]
```diff
@@ -1,25 +1,20 @@
...
 import { Theme } from "../styles";
 import { ResultRendererProps, Results } from "./Results";
-import { isNotEmptyArray, isNotEmptyString } from "./shared/utils";
 import { Suggestions } from "./Suggestions";
 import { Typeahead } from "./Typeahead";
-
-const ReturnKeyCode = 13;
-const RightArrowKeyCode = 39;
+import { Search } from "../Search";
+import {
```
[the remark this answers is not in the document]
Comment (zlatanpham) [reply src/components/Input/Input.tsx]: Hey @benhinchley, you mean  `PipelineProps`, etc should be exported from `/Search/index.ts`? So we can do:

```js
import { Search, PipelineProps, SearchState, StateChangeOptions } from "../Search"
```

File: src/components/Search/search.mdx (hunk 1 above)
Comment (zlatanpham) [reply src/components/Search/search.mdx]: I've updated the styling. Let me know if having any issue.

File: src/components/Input/Input.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (benhinchley) [reply src/components/Input/Input.tsx]: @zlatanpham yep exactly, just a slightly nicer developer experience.

[the remark this answers is not in the document]
Comment (zlatanpham) [reply src/components/Input/Input.tsx]: done!

**Pull request merged**